pkg/util/util.go

```diff
@@ -25,11 +26,47 @@ func GetCertPool(paths []string) (*x509.CertPool, error) {
 	return pool, nil
 }
 
+// GetRequestProto return the request host header or X-Forwarded-Proto if present
+func GetRequestProto(req *http.Request) string {
+	proto := req.Header.Get("X-Forwarded-Proto")
+	if proto == "" {
+		proto = req.URL.Scheme
+	}
+	return strings.ToLower(proto)
+}
+
 // GetRequestHost return the request host header or X-Forwarded-Host if present
 func GetRequestHost(req *http.Request) string {
 	host := req.Header.Get("X-Forwarded-Host")
 	if host == "" {
 		host = req.Host
 	}
-	return host
+	return strings.ToLower(host)
+}
+
+// GetRequestPortFromHeader return the request port from X-Forwarded-Port if present, if not; use provided port
```

Comment:
I think based on the extended comments around `X-Forwarded-Port`, we came to the conclusion this is superfluous and should be removed -- it will be in the `X-Forwarded-Host`?

I'd prefer we remove the complexity and we can tack on `X-Forwarded-Port` if someone truly finds a use case for it when that arises.

Since this isn't a header most proxies aren't sending, I'm not sure what implications arise if a client messes with things and injects one. Hence, I'd prefer we don't use or trust it for now.

Reply:
Alright, will remove the port header check.